robot_skills/src/robot_skills/lights.py

```diff
@@ -39,45 +34,58 @@ def set_color(self, r, g, b, a=1.0):
         :param a: alpha value 0.0-1.0
         :return: no return
         """
-        self.set_color_colorRGBA(ColorRGBA(r, g, b, a))
+        self.set_color_rgba_msg(ColorRGBA(r, g, b, a))
 
-    def set_color_colorRGBA(self, rgba):
+    def set_color_rgba_msg(self, rgba):
         """
         Set the color of the robot by a std_msgs.msg.ColorRGBA
 
         :param rgba: std_msgs.msg.ColorRGBA
         :return: no return
         """
-        rgb_msg = RGBLightCommand(color=rgba)
-        rgb_msg.show_color.data = True
-        self._topic.publish(rgb_msg)
+        self._send_color_msg(rgba_msg=rgba)
+
+    def _send_color_msg(self, rgba_msg):
+        """
+        Sends the color message to the robot hardware. This function needs to be implemented by deriving classes.
+
+        :param rgba_msg: message to send
+        """
+        # ToDo: replace by fstring (after going to Python3)
+        raise NotImplementedError("_send_color_msg is not implemented for {}".format(self.__class__.__name__))
 
     def selfreset(self):
         """
         Set the lights to blue
 
         :return: no return
         """
-        self.set_color_colorRGBA(RESET)
+        self.set_color_rgba_msg(RESET)
         return True
 
-    def on(self):
+
+class TueLights(LightsInterface):
+    def __init__(self, robot_name, tf_listener):
         """
-        Set the lights of the robot ON
+        Interface to the robot's lights. This uses the TU/e-specific RGBLightCommand message type.
 
-        :return: no return
+        :param robot_name: robot_name
+        :param tf_listener: tf_server.TFClient()
         """
-        rgb_msg = RGBLightCommand(show_color=True)
-        self._topic.publish(rgb_msg)
+        super(TueLights, self).__init__(robot_name=robot_name, tf_listener=tf_listener)
+        self._publisher = rospy.Publisher(
+            '/'+robot_name+'/rgb_lights_manager/user_set_rgb_lights', RGBLightCommand, queue_size=10
```

Comment:
Use format instead of string appending?

Reply:
I'd be happy to change it but not sure why. In any way, in Python 3 we can use f-strings

Reply:
Yes, I prefer it that way. Plus it is a little bit faster I think. F strings have to wait a month

Reply:
Not sure if it's faster (see [StackOverflow](https://stackoverflow.com/questions/38722105/format-strings-vs-concatenation)) I changed it in 0773c51f963052903cd1ff4b008659b51f45d874